### ISSUE TITLE
permissions errors on working directory when multiple containers attach the same volumes.

### ISSUE THREAD

**Issue by rustysys-dev** [containers/podman#7512]
/kind bug

**Description**

tl;dr : When multiple containers attach to the same volumes, only the last container has access to the working directory when working directory within said volume.

For local troubleshooting of a production application we have a docker-compose file for which I use podman-compose to bring up the test environment. 

I created a test that can reproduce the error below.

**Steps to reproduce the issue:**

1.  Either use the above `docker-compose.yaml` or the above podman commands to reproduce the environment

podman-compose:

```
podman-compose up -d
```

docker-compose.yaml

```yaml
version: '3.7'

services:
  auth_dev: &app
    image: "alpine:latest"
    volumes:
      - .:/test:Z
    working_dir: /test/wrkdir
    ports:
      - 8081:8081
    command: 'sleep 1000000'
  mock-asset-api:
    <<: *app
    ports:
      - 8082
  mock-search-api:
    <<: *app
    ports:
      - 8083
  mock-event-api:
    <<: *app
    ports:
      - 8084
```

podman:

```
podman pod create --name=test --share net -p 8084 -p 8082 -p 8083 -p 8081:8081
a3fb02591d068b751fbd4d0ceda56b511d6eae711a7e0dac8c91ca39bd935eb7
0
podman run --name=test_auth_dev_1 -d --pod=test --label io.podman.compose.config-hash=123 --label io.podman.compose.project=test --label io.podman.compose.version=0.0.1 --label com.docker.compose.container-number=1 --label com.docker.compose.service=auth_dev --mount type=bind,source=/home/rustysysdev/test/.,destination=/test,bind-propagation=Z --add-host auth_dev:127.0.0.1 --add-host test_auth_dev_1:127.0.0.1 --add-host mock-asset-api:127.0.0.1 --add-host test_mock-asset-api_1:127.0.0.1 --add-host mock-search-api:127.0.0.1 --add-host test_mock-search-api_1:127.0.0.1 --add-host mock-event-api:127.0.0.1 --add-host test_mock-event-api_1:127.0.0.1 -w /test/wrkdir alpine:latest sleep 1000000
d2fc99113cf3866645548907a9820a8e72edc437905af899c034806efc61cb44
0
podman run --name=test_mock-asset-api_1 -d --pod=test --label io.podman.compose.config-hash=123 --label io.podman.compose.project=test --label io.podman.compose.version=0.0.1 --label com.docker.compose.container-number=1 --label com.docker.compose.service=mock-asset-api --mount type=bind,source=/home/rustysysdev/test/.,destination=/test,bind-propagation=Z --add-host auth_dev:127.0.0.1 --add-host test_auth_dev_1:127.0.0.1 --add-host mock-asset-api:127.0.0.1 --add-host test_mock-asset-api_1:127.0.0.1 --add-host mock-search-api:127.0.0.1 --add-host test_mock-search-api_1:127.0.0.1 --add-host mock-event-api:127.0.0.1 --add-host test_mock-event-api_1:127.0.0.1 -w /test/wrkdir alpine:latest sleep 1000000
374a59c3044549dbcc3146c985ffedb5843f2393d576e06530bd9c78323cfd25
0
podman run --name=test_mock-search-api_1 -d --pod=test --label io.podman.compose.config-hash=123 --label io.podman.compose.project=test --label io.podman.compose.version=0.0.1 --label com.docker.compose.container-number=1 --label com.docker.compose.service=mock-search-api --mount type=bind,source=/home/rustysysdev/test/.,destination=/test,bind-propagation=Z --add-host auth_dev:127.0.0.1 --add-host test_auth_dev_1:127.0.0.1 --add-host mock-asset-api:127.0.0.1 --add-host test_mock-asset-api_1:127.0.0.1 --add-host mock-search-api:127.0.0.1 --add-host test_mock-search-api_1:127.0.0.1 --add-host mock-event-api:127.0.0.1 --add-host test_mock-event-api_1:127.0.0.1 -w /test/wrkdir alpine:latest sleep 1000000
f2421c17d60c3b9d4e6b4efc2a1784189555658398a42cc34f6012e94edd4498
0
podman run --name=test_mock-event-api_1 -d --pod=test --label io.podman.compose.config-hash=123 --label io.podman.compose.project=test --label io.podman.compose.version=0.0.1 --label com.docker.compose.container-number=1 --label com.docker.compose.service=mock-event-api --mount type=bind,source=/home/rustysysdev/test/.,destination=/test,bind-propagation=Z --add-host auth_dev:127.0.0.1 --add-host test_auth_dev_1:127.0.0.1 --add-host mock-asset-api:127.0.0.1 --add-host test_mock-asset-api_1:127.0.0.1 --add-host mock-search-api:127.0.0.1 --add-host test_mock-search-api_1:127.0.0.1 --add-host mock-event-api:127.0.0.1 --add-host test_mock-event-api_1:127.0.0.1 -w /test/wrkdir alpine:latest sleep 1000000
c58296ab0bd5b159029ae7960453519306943a12009afa8a88a3c5454516593b
0
```

2.  try to `ls` for each container and see the last one started is able to perform `ls`

```
[rustysysdev@localhost test]$ podman exec -ti test_auth_dev_1 ls
ls: .: Permission denied
Error: exec session exited with non-zero exit code 1: OCI runtime error
[rustysysdev@localhost test]$ podman exec -ti test_mock-asset-api_1 ls
ls: .: Permission denied
Error: exec session exited with non-zero exit code 1: OCI runtime error
[rustysysdev@localhost test]$ podman exec -ti test_mock-event-api_1 ls
emptyfile.yaml
[rustysysdev@localhost test]$ podman exec -ti test_mock-search-api_1 ls
ls: .: Permission denied
Error: exec session exited with non-zero exit code 1: OCI runtime error 
```

**Describe the results you received:**

Permission denied for all but the last contianer started.

**Describe the results you expected:**

ls output for each container.

**Additional information**

**security context of directory:**

```
[rustysysdev@localhost test]$ ls -laZ
total 4
drwxr-xr-x. 1 rustysysdev hugetlbfs system_u:object_r:container_file_t:s0:c852,c870  50 Aug 31 23:41 .
drwx------. 1 rustysysdev hugetlbfs unconfined_u:object_r:user_home_dir_t:s0        762 Aug 31 23:41 ..
-rw-r--r--. 1 rustysysdev hugetlbfs system_u:object_r:container_file_t:s0:c852,c870 357 Aug 31 23:41 docker-compose.yaml
drwxr-xr-x. 1 rustysysdev hugetlbfs system_u:object_r:container_file_t:s0:c852,c870  28 Aug 31 23:42 wrkdir
```

**Output of `podman version`:**

```
[rustysysdev@localhost auth-gateway]$ podman version
Version:      2.0.5
API Version:  1
Go Version:   go1.14.6
Built:        Thu Jan  1 09:00:00 1970
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

```
[rustysysdev@localhost test]$ podman info --debug
host:
  arch: amd64
  buildahVersion: 1.15.1
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.19-1.fc32.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.19, commit: 5dce9767526ed27f177a8fa3f281889ad509fea7'
  cpus: 4
  distribution:
    distribution: fedora
    version: "32"
  eventLogger: file
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.8.4-200.fc32.x86_64
  linkmode: dynamic
  memFree: 9002905600
  memTotal: 16536584192
  ociRuntime:
    name: crun
    package: crun-0.14.1-3.fc32.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.14.1
      commit: 598ea5e192ca12d4f6378217d3ab1415efeddefa
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.4-1.fc32.x86_64
    version: |-
      slirp4netns version 1.1.4
      commit: b66ffa8e262507e37fca689822d23430f3357fe8
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 2
  swapFree: 8336175104
  swapTotal: 8336175104
  uptime: 1h 53m 24.11s (Approximately 0.04 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/rustysysdev/.config/containers/storage.conf
  containerStore:
    number: 5
    paused: 0
    running: 5
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.1.2-1.fc32.x86_64
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  graphRoot: /home/rustysysdev/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 171
  runRoot: /run/user/1000/containers
  volumePath: /home/rustysysdev/.local/share/containers/storage/volumes
version:
  APIVersion: 1
  Built: 0
  BuiltTime: Thu Jan  1 09:00:00 1970
  GitCommit: ""
  GoVersion: go1.14.6
  OsArch: linux/amd64
  Version: 2.0.5
```

**Package info:**

```
[rustysysdev@localhost auth-gateway]$ rpm -q podman
podman-2.0.5-1.fc32.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details:**

Fedora 32

**Comment by Luap99** [comment]
From the podman create man page
> To change a label in the container context, you can add either of two suffixes :z or :Z to the volume mount. These suffixes tell Podman to relabel file objects on the shared volumes. The z option tells Podman that two containers share the volume content. As a result, Podman labels the content with a shared content label. Shared volume labels allow all containers to read/write content. The Z option tells Podman to label the content with a private unshared label. Only the current container can use a private volume.

so you need to you a lower case `z`

**Comment by rustysys-dev** [comment]
> From the podman create man page
> 
> > To change a label in the container context, you can add either of two suffixes :z or :Z to the volume mount. These suffixes tell Podman to relabel file objects on the shared volumes. The z option tells Podman that two containers share the volume content. As a result, Podman labels the content with a shared content label. Shared volume labels allow all containers to read/write content. The Z option tells Podman to label the content with a private unshared label. Only the current container can use a private volume.
> 
> so you need a lower case `z`

just saw that... sorry closing!
